### PR TITLE
alacritty: new package

### PR DESCRIPTION
**Pull request by ognevnydemon**
Alacritty - a cross-platform, OpenGL terminal emulator.

Successfully builds, but I can't test it now.
@jeremyd2019 any thoughts on CLANGARM64 compatibility?

### CODE REVIEW

**Comment by ognevnydemon** [comment]
why check() is [skipped](https://github.com/msys2/MINGW-packages/actions/runs/5625465360/job/15244587292)?

**Comment by Biswa96** [comment]
Checks are disabled for CI because those take too much time or failed most of the time. See the `--nocheck` option in `.ci/ci-build.sh` file.

**Comment by ognevnydemon** [comment]
are there any plans to enable it?

**Comment by Biswa96** [comment]
> are there any plans to enable it?

Nope. It is assumed that package maintainers/contributors did some rudimentary tests.

File: mingw-w64-alacritty/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+              'A56EF308A9F1256C25ACA3807EA8F8B94622A6A9') # Kirill Chibisov <contact@kchibisov.com>
+sha256sums=('SKIP')
+
+prepare() {
+  cd "${srcdir}/${_realname}"
+
+  if [[ $MINGW_PACKAGE_PREFIX == *-clang-aarch64 ]]; then
+    cargo fetch --locked --target aarch64-pc-windows-gnullvm
+  else
+    cargo fetch --locked --target "${CARCH}-pc-windows-gnu"
```
Comment (jeremyd2019) [inline mingw-w64-alacritty/PKGBUILD:34]: I don't remember any other package needing to specify a target to cargo like this.  Why is this necessary?

**Comment by jeremyd2019** [comment]
> Successfully builds, but I can't test it now. @jeremyd2019 any thoughts on CLANGARM64 compatibility?

Do you just want me to test it?  Actually running opengl things on arm64 is kind of interesting (needs a compatibility store app thing from Microsoft).

File: mingw-w64-alacritty/PKGBUILD (hunk 1 above)
Comment (ognevnydemon) [reply mingw-w64-alacritty/PKGBUILD:34]: I find that it uses lots of unneeded dependencies (~230), after this this number has been greatly reduced (~110)

**Comment by ognevnydemon** [comment]
> > Successfully builds, but I can't test it now. @jeremyd2019 any thoughts on CLANGARM64 compatibility?
> 
> Do you just want me to test it?  Actually running opengl things on arm64 is kind of interesting (needs a compatibility store app thing from Microsoft). 

yes, why not? I don't have machine for this, so how can I know?

File: mingw-w64-alacritty/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+pkgdesc="A cross-platform, OpenGL terminal emulator (mingw-w64)"
+arch=('any')
+mingw_arch=('mingw32' 'mingw64' 'ucrt64' 'clang32' 'clang64' 'clangarm64')
+url="https://github.com/alacritty/alacritty"
+license=('spdx:Apache-2.0 OR MIT')
+makedepends=("${MINGW_PACKAGE_PREFIX}-rust"
+             "${MINGW_PACKAGE_PREFIX}-cmake"
+             "${MINGW_PACKAGE_PREFIX}-ncurses"
+             "${MINGW_PACKAGE_PREFIX}-desktop-file-utils"
+             "${MINGW_PACKAGE_PREFIX}-gdb"
```
Comment (jeremyd2019) [inline mingw-w64-alacritty/PKGBUILD:17]: Why is gdb needed to build?  There is no gdb package for arm64, so it fails here right now.

Comment (ognevnydemon) [reply mingw-w64-alacritty/PKGBUILD:17]: hmm, didn't find gdb in upstream, but only in arch's package. maybe we can try to remove it

**Comment by jeremyd2019** [comment]
bunch of errors building on arm64:
```
  error[E0412]: cannot find type `ULONG_PTR` in this scope
    --> C:\Users\Administrator\.cargo\registry\src\index.crates.io-6f17d22bba15001f\winapi-0.2.8\src\basetsd.rs:86:19
     |
  86 | pub type SIZE_T = ULONG_PTR;
     |                   ^^^^^^^^^ not found in this scope
  
```

and a bunch of similar in winapi-0.2.8, having to do with pointer-sized integral types.  perhaps that version is too old for windows on arm64 support?

**Comment by ognevnydemon** [comment]
> perhaps that version is too old for windows on arm64 support?

maybe. it seems that upstream updated deps, but hasn't released this yet

**Comment by jeremyd2019** [comment]
Yes, winapi 0.2 has only x86 and x86_64 definitions of sized types in https://github.com/retep998/winapi-rs/blob/0.2/src/basetsd.rs.  Whereas 0.3 got arm support (both arm64 and arm32), and https://github.com/retep998/winapi-rs/blob/0.3/src/shared/basetsd.rs refers instead to `target_pointer_width` cfg attr.  Unfortunately, alacritty master Cargo.lock still refers to winapi 0.2.8, apparently via mio 0.6.23?

**Comment by ognevnydemon** [comment]
I opened issue in upstream and I received this answer: https://github.com/alacritty/alacritty/issues/7104#issuecomment-1646640836
